lib/util/version.js

```diff
@@ -14,11 +14,17 @@ function resetWarningFlag() {
   warnedForMissingVersion = false;
 }
 
+let cachedDetectedReactVersion;
 function detectReactVersion() {
+  if (cachedDetectedReactVersion) {
+    return cachedDetectedReactVersion;
+  }
+
   try {
     const reactPath = resolve.sync('react', {basedir: process.cwd()});
     const react = require(reactPath); // eslint-disable-line global-require, import/no-dynamic-require
-    return react.version;
+    cachedDetectedReactVersion = react.version;
```

Comment:
let's also assign `cachedDetectedReactVersion` on line 34.5, so that the error case is cached too?

Reply:
Done, I also fixed the tests I think